lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -8537,18 +8537,18 @@ using namespace Js;
         CallFlags flags = CallFlags_Value;
         Var element = nullptr;
         Var testResult = nullptr;
-        int64 loopStart = reversed ? length - 1 : 0;
+        uint32 loopStart = reversed ? (uint32)length - 1 : 0;
         int8 loopDelta = reversed ? -1 : 1;
 
         if (pArr)
         {
             Var undefined = scriptContext->GetLibrary()->GetUndefined();
 
             Assert(length <= UINT_MAX);
-            for (int64 k = loopStart; reversed ? k >= 0 : k < length; k += loopDelta)
+            for (uint32 k = loopStart; (!reversed || k >= 0) && k < length; k += loopDelta)
```

Comment:
Instead of `(!reversed || k >= 0) && k < length` can you just put `k < length` isn't it equivalent?

Reply:
Oh.....Yep. It's soooo tricky...